File: MAP/Lab7MAP/src/domain/PhdStudent.java
package domain;

public class PhdStudent extends Student implements ComparableObject<Student>,Identifiable {
	protected String thesis;
	protected String supervisor;
	protected int grade2;
	protected int grade3;
	
	public PhdStudent(int id, String name,String thesis, String supervisor, int grade,int grade2,int grade3){
		super(id,name,grade);
		this.thesis=thesis;
		this.grade2=grade2;
		this.grade3=grade;
	}
	
	@Override
	public float getAverage(){
		return (grade+grade2+grade3)/3;
	}
	
	@Override
	public boolean isGreaterThan(Student s){
		return this.getAverage()>s.getAverage();
	}
	
	@Override
	public String toString(){
		return +id+" "+name+" "+supervisor+" "+thesis+" "+grade+" "+grade2+" "+grade3;
	}

}

File: MAP/Csharp lab2/Csharp lab2/Domain/PhdStudent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Csharp_lab2.Domain
{
    [Serializable]
    public class PhdStudent : Student, Identifiable, ComparableObject<Student>
    {
        public String supervisor;
        public String thesis;
        public int grade2;
        public int grade3;
        public PhdStudent(int id, String name, String supervisor, String thesis, int grade, int grade2,int grade3):base (id, name, grade) {
                                this.supervisor = supervisor;
                                this.thesis = thesis;
                                this.grade2 = grade2;
                                this.grade3 = grade3;
                        }

                        public override float getAverage() {
                                return (this.grade + this.grade2+this.grade3) / 2;
                        }

                        public bool isGreaterThan(Student student) {
                                return (this.getAverage() > this.getAverage());
                        }


                        public override String ToString()
                        {
                            //Override for the string output function
                            //Precond:- function called by a student object
                            //Postcont:- outputs the object values in a string format

                            return id + " " + supervisor + " " + thesis + " " + name + " " + grade + " " + grade2 + " " + grade3;
                        }
    }
}
